### PR TITLE
Improved deletion in Meta

### PR DESCRIPTION
**Pull request by aburrell**
# Description

Partially addresses #1078 by adding `__delitem__` to Meta.  Also adds `drop` methods to MetaLabels and MetaHeader.  Further allows MetaHeater attributes to be acceessed directly from Meta.

## Type of change

Please delete options that are not relevant.

- New feature (non-breaking change which adds functionality)
- This change requires a documentation update

# How Has This Been Tested?

Added new unit tests.  Also:

```
inst = pysat.Instrument('pysat', 'testing')
inst.load(date=pysat.instrument.pysat_testing._test_date[''][''], use_header=True)

# Add high-level metadata
inst.meta.header.PI = "Me!"

# Delete meta data for data variable
del inst.meta['uts']

# Delete meta data of a specific type
del inst.meta['units']

# Delete the header data
del inst.meta['PI']

print(inst.meta)
```

**Test Configuration**:
* Operating system: OS X big sur
* Version number: Python 3.8, 3.9
* Any details about your local setup that are relevant

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``main``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes

If this is a release PR, replace the first item of the above checklist with the release
checklist on the wiki: https://github.com/pysat/pysat/wiki/Checklist-for-Release

### CODE REVIEW

File: pysat/_meta.py [hunk 1]
```diff
@@ -2029,6 +2081,37 @@ def default_values_from_attr(self, attr_name, data_type=None):
 
         return default_val
 
+    def drop(self, names):
+        """Removes data from MetaLabels.
```
Comment (aburrell) [inline pysat/_meta.py:2085]:
```suggestion
        """Remove data from MetaLabels.
```

File: pysat/tests/test_meta_labels.py [hunk 2]
```diff
@@ -185,6 +186,24 @@ def test_default_value_from_type_int_inputs(self, in_val, comp_val):
 
         return
 
+    @pytest.mark.parametrize("drop_labels", [["units", "fill_val"], "units"])
+    def test_drop(self, drop_labels):
+        """Test successfull drop from MetaLabels.
```
Comment (aburrell) [inline pysat/tests/test_meta_labels.py:191]:
```suggestion
        """Test successful drop from MetaLabels.
```

**Comment by jklenzing** [comment]
@aburrell, I'm done with my round of deprecation removals. Hopefully this helps create a cleaner environment here

File: pysat/_meta.py [hunk 3]
```diff
@@ -798,13 +801,37 @@ def drop(self, names):
...
+            header data
 
+        """
+        # Ensure the input is list-like
+        names = listify(names)
+        for name in names:
+            if name in self._ho_data:
+                # Drop the higher dimension data
+                self._ho_data.pop(name)
+
```
Comment (jklenzing) [inline pysat/_meta.py:821]:
```suggestion
```

**Review by jklenzing — COMMENTED** [review]
The develop merge is failing because this branch adds checks for the higher order data, which was removed.

***EDIT: verifed that tests pass locally when these lines are removed.***

File: pysat/_meta.py (hunk 3 above)
Comment (aburrell) [reply pysat/_meta.py:821]: This can only be applied after #1148 is merged in.

**Review by jklenzing — APPROVED** [review]

**Pull request merged**